src/dc_configure.rs

```diff
@@ -15,15 +15,10 @@ use crate::x::*;
 
 macro_rules! progress {
     ($context:tt, $progress:expr) => {
+        assert!($progress >= 0 && $progress <= 1000);
```

Comment:
how about you add a nice error message to explain what is wrong?

Reply:
did so :)